src/main/java/org/threeten/extra/MutableClock.java

```diff
@@ -0,0 +1,511 @@
+ * <li>Verify some behavior of the component in the initial state
+ * <li>Advance the clock <em>without recreating the component</em>
+ * <li>Verify that the component behaves as expected given the (artificial)
+ *     delta in clock time since the initial state
+ * </ol>
+ * <p>
+ * This class is mutable. The time-zone of the clock is fixed, but the instant
+ * may be updated at will.
+ * <p>
+ * The instant may be set to any value even if that new value is less than the
+ * previous value. Caution should be exercised when moving the clock backwards,
+ * since clock-sensitive components are likely to assume that time is
+ * monotonically increasing.
+ * <p>
+ * Update semantics are expressed in terms of {@link ZonedDateTime}. The steps
+ * of each update are as follows:
+ * <ol>
+ * <li>The clock captures its own state in a {@code ZonedDateTime} via {@link
+ *     ZonedDateTime#now(Clock)} (or the equivalent thereof)
+ * <li>The update operation is applied to that {@code ZonedDateTime}, producing
+ *     a new {@code ZonedDateTime}
+ * <li>The resulting {@code ZonedDateTime} is converted to an instant via {@link
+ *     ZonedDateTime#toInstant()} (or the equivalent thereof)
+ * <li>The clock's instant is set to that new instant
+ * </ol>
+ * <p>
+ * Therefore, whenever there is a question about what argument types, units,
+ * fields, or values an update operation supports, or what the result will be,
+ * refer to the corresponding method of {@code ZonedDateTime}. Links are
+ * provided from the documentation of each update operation of this class to the
+ * corresponding method of {@code ZonedDateTime}.
+ * <p>
+ * While update semantics are expressed in terms of {@code ZonedDateTime}, that
+ * imposes no requirements on implementation details. The implementation may
+ * avoid using {@code ZonedDateTime} completely or only sometimes, for
+ * convenience, efficiency, or any other reason.
+ *
+ * <h3>Implementation Requirements:</h3>
+ * This class is thread-safe. Updates are atomic and synchronized.
+ *
+ * @serial exclude
+ */
+public final class MutableClock
+        extends Clock
+        implements Serializable {
+
+    /**
+     * Serialization version.
+     */
+    private static final long serialVersionUID = -6152029959790119695L;
+
+    /**
+     * The mutable instant of this clock.
+     */
+    private final InstantHolder instantHolder;
+
+    /**
+     * The fixed time-zone of this clock.
+     */
+    private final ZoneId zone;
+
+    /**
+     * Obtains a new {@code MutableClock} set to the epoch of
+     * 1970-01-01T00:00:00Z, converting to date and time using the UTC
+     * time-zone.
+     * <p>
+     * Use this method when a {@code MutableClock} is needed and neither its
+     * initial value nor its time-zone are important.
+     *
+     * @return a new {@code MutableClock}, not null
+     */
+    public static MutableClock epochUTC() {
+        return MutableClock.of(EPOCH, UTC);
+    }
```

Comment:
It's the only factory that I use anymore!

The use case is testing what happens after some relative amount of time has elapsed.  That code tends to be shaped like this:

```java
MutableClock clock = MutableClock.epochUTC();
ThingToTest component = new ThingToTest(clock);
component.doSomething();
clock.add(periodOrDuration);
assertEquals(component.getResult(), ...);
```

In those cases, the initial instant and time-zone don't matter theoretically.  Therefore, I should:

* Pick fixed arguments that are the same for everyone running the test.  It's good if the arguments don't depend on wall clock time or the default time-zone, since it makes the test output as consistent as possible.
* Make that choice of arguments only once.  Encode it in a factory method so I don't have to think about it or spell out the arguments each time.

And I do think it's worth encouraging other people to do the same.